Gobabygo/Gobabygo.ino

```diff
@@ -41,25 +43,39 @@ void loop() {
   scaledJoyXVal = joyXVal/1023*2 - 1;
   correctForDeadzone(scaledJoyXVal, deadzone);
   
-  ledVal = map(joyXVal, 0, 1023, 0, 255);
-  analogWrite(rightMotorPin, ledVal);
+  // ledVal = map(joyXVal, 0, 1023, 0, 255);
+  // analogWrite(rightMotorPin, ledVal);
   
   joyYVal = analogRead(joyStickYPin);
   scaledJoyYVal = 2/1023 * joyYVal -1;
   correctForDeadzone(scaledJoyYVal, deadzone);
 
-  // "Loop" that happens twice every second
+  // Cheesy drive logic
+  leftMotorVal = scaledJoyYVal + scaledJoyXVal;
+  rightMotorVal = scaledJoyYVal - scaledJoyXVal;
+
+  if (abs(leftMotorVal) > 1 or abs(rightMotorVal) > 1) {  // the value of either motor can never be outside the range -1 to 1
+    scaler = max(abs(leftMotorVal), abs(rightMotorVal));
+    leftMotorVal = leftMotorVal / scaler;
+    rightMotorVal = rightMotorVal / scaler;
+  }
+
+  analogWrite(leftMotorPin, leftMotorVal * 127.5 + 127.5);    // converting from -1-1 to 0-255
+  analogWrite(rightMotorPin, rightMotorVal * 127.5 + 127.5);
```

Comment:
Consider using a the map() function here as that might be more clear.

Reply:
The map function only works for integers, so here it would make the motor values only have three possible values, -1, 0, and 1.